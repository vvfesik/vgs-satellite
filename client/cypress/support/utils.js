export const cutLine = (source, lineString = 'name: ') =>
  source
    .split('\n')
    .filter(line => !line.includes(lineString))
    .join('\n');
