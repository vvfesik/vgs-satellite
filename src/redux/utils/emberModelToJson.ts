function emberModelToJson(route: object | null) {
  let json: object | null;

  try {
    json = JSON.parse(JSON.stringify(route));
  } catch (error) {
    json = error.message;
  }

  return json;
}

export default emberModelToJson;
