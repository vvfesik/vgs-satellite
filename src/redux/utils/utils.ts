import moment from 'moment';
import classnames from 'classnames';
import parseUrl from 'url-parse';
import randomId from './random-id';
import { volatileRulesRegex } from 'src/data/regex';
import { includes } from 'lodash';
import { ILog } from 'src/redux/interfaces/logs';

export function formatDate(date) {
  if (moment().isSame(date, 'day')) {
    return moment.utc(date).local().format('[Today] HH:mm:ss');
  } else {
    return moment.utc(date).local().format('YYYY-MM-DD HH:mm:ss');
  }
}

export function unixToFormat(dateUnix: number, format?: string) {
  return moment.unix(dateUnix).format(format);
}

export function dateFromNow(date) {
  return moment(date).fromNow();
}

export function nowBeforeDate(date) {
  return moment().isBefore(date);
}

export function isVolatile(name) {
  return name.match(volatileRulesRegex);
}

export function badgeColor(statusCode) {
  return classnames({
    danger: statusCode >= 500,
    success: statusCode >= 200 && statusCode <= 299,
    secondary: statusCode >= 300 && statusCode <= 399,
    warning: statusCode >= 400 && statusCode <= 499,
  });
}

export function isEqual(stringOne: string, stringTwo: string) {
  return stringOne.toLowerCase() === stringTwo.toLowerCase();
}

export function getQueryParameters(url: string) {
  return parseUrl(url).query;
}

export const generateRouteName = (hostname?: string) => {
  const routeName = randomId();
  return routeName;
};

export function constructUriFromLog(log: ILog) {
  const isPathSufficient = !!log.path?.match(/^(https?:)\/\//);
  const isPathAppendable = !log.upstream?.match(/\/$/) && !!log.path?.match(/^\//);
  const isPathPrependable = !!log.scheme && !log.upstream.startsWith(`${log.scheme}:`) && !log.upstream?.match(/^\//);
  return isPathSufficient
    ? log.path
    : `${isPathPrependable ? log.scheme + '://' : ''}${log.upstream ?? ''}${isPathAppendable ? log.path : ''}`;
}

export const getFiltersWithoutOperations = (route: IRoute) => route.entries.filter(entry => !entry.operations);

export function isValidHostname(hostname: string) {
  const hostnameRegex = /^\s*((?=.{1,255}$)[0-9A-Za-z](?:(?:[0-9A-Za-z]|\b-){0,61}[0-9A-Za-z])?(?:\.{1,}[0-9A-Za-z](?:(?:[0-9A-Za-z]|\b-){0,61}[0-9A-Za-z])?){1,}?)\s*$/; // tslint:disable-line
  return hostnameRegex.test(hostname);
}

export const hostFromUri = (destOverride: string) => {
  const url = new URL(destOverride);
  return url.hostname;
};

export const hostnameFromUri = (destOverride: string) => destOverride.replace(/^(.*\/\/)/, '');

export function removeQueryParameters(url: string) {
  const queryParameter = parseUrl(url).query;
  return url.replace(queryParameter, '');
};

export const isOperatorExist = (rules: any, operator: string) => includes(rules.map(rule => rule.operator), operator);

export function isJSON(data) {
  try {
    JSON.parse(data);
    return true;
  } catch {
    return false;
  }
}

export function getOperationsName(operations: string) {
  const operationsJson = isJSON(operations) && JSON.parse(operations);
  if (Array.isArray(operationsJson)) {
    const operationsArray = operationsJson.map(op => op['@type']);
    return operationsArray.join(', ').replace(/(type\.googleapis\.com\/)|(Config)/g, '');
  } else {
    return '';
  }
}

export function isRegExp(value: string) {
  if (!value) {
    return true;
  }
  try {
    const regExp = new RegExp(value);
    return true;
  } catch (e) {
    return false;
  }
}
