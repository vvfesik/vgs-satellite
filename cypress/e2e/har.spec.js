import { cutLine } from '../support/utils';

describe('Localhoste upload single har flow', function() {
  beforeEach(() => {
    cy.server();
    cy.route('GET', 'flows.json').as('getFlows');
    cy.route('GET', 'route').as('getRoutes');
    cy.cleanupFlows();
    cy.cleanupRoutes();
    cy.fixCypressSpec(__filename);
  });

  it('Visits Localhoste and gets 2 yamls from uploaded har file', function() {
    cy.visit('/');
    cy.wait(['@getRoutes', '@getFlows']);
    cy.get('[data-role="import-from-har"]').attachFile('upload-post.har');

    cy.contains('/post');

    cy.get('[data-role="logs-row"]').click();
    cy.contains('http://httpbin.org/post');
    cy.get('[data-role="log-details-modal"]').toMatchSnapshot();

    cy.get('[data-role="select-response-phase"]').click();
    cy.get('[data-role="select-request-phase"]').click();

    cy.get('[data-role="tab-headers"]').click();
    cy.get('[data-role="log-details-modal"]').toMatchSnapshot();

    cy.get('[data-role="select-response-phase"]').click();
    cy.get('[data-role="log-details-modal"]').toMatchSnapshot();
    cy.get('[data-role="select-request-phase"]').click();

    cy.get('[data-role="tab-body"]').click();
    cy.get('[data-role="log-details-modal"]').toMatchSnapshot();

    cy.get('[data-role="select-response-phase"]').click();
    cy.get('[data-role="log-details-modal"]').toMatchSnapshot();
    cy.get('[data-role="select-request-phase"]').click();
    cy.contains('{"foo": "bar"}');

    cy.get('.ant-btn-primary').click();
    cy.get('[data-role="quick-integration-modal"]').toMatchSnapshot();
    cy.contains('foo: bar').click({ force: true });

    cy.get('[data-role="select-secure-payload"]').click();
    cy.get('.ant-modal-content .ant-modal-title').contains('Save route');
    cy.get('.ant-modal-content .nav-tabs .active').contains('Inbound');
    cy.get('.ant-modal-content .active [data-role="inbound-code-container"]');

    cy.fixture('upload-inbound.yaml').then(yaml => {
      cy.get('[data-role="inbound-code-container"] pre code').should($div => {
        expect(cutLine($div.get(0).innerText)).to.eq(cutLine(yaml));
      });
    });
    cy.contains('Export Inbound YAML').click();

    cy.get('.nav-link')
      .contains('Outbound')
      .click();

    cy.get('.ant-modal-content .nav-tabs .active').contains('Outbound');
    cy.get('.ant-modal-content .active [data-role="outbound-code-container"]');

    cy.fixture('upload-outbound.yaml').then(yaml => {
      cy.get('[data-role="outbound-code-container"] pre code').should($div => {
        expect(cutLine($div.get(0).innerText)).to.eq(cutLine(yaml));
      });
    });
    cy.contains('Export Outbound YAML').click().type('{esc}');
  });
});
