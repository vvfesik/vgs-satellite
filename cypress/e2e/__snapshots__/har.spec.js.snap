exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #0`] = `
<div
  data-role="log-details-modal"
  class="modal-dialog modal-lg flow-detail"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Secure Data</h5>
      <span id="" class="cursor-pointer"><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <nav class="nav-tabs nav-tabs-sm">
        <a href="#" class="active capitalize" data-role="tab-general">general</a
        ><a href="#" class="capitalize" data-role="tab-headers">headers</a
        ><a href="#" class="capitalize" data-role="tab-body">body</a>
      </nav>
      <div class="text-center mt-3 mb-2 position-relative">
        <button
          type="button"
          class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
        >
          <span>Secure this payload</span>
        </button>
        <div role="group" class="btn-group">
          <button
            type="button"
            data-role="select-request-phase"
            class="active capitalize btn btn-primary-light btn-sm"
          >
            request</button
          ><button
            type="button"
            data-role="select-response-phase"
            class="capitalize btn btn-outline btn-sm"
          >
            response
          </button>
        </div>
        <hr class="my-3 p-0" />
      </div>
      <div class="mb-3 pt-0">
        <table class="general-table pb-3 table">
          <tbody>
            <tr>
              <td class="smallbold">Protocol</td>
              <td><span>HTTP/1.1</span></td>
            </tr>
            <tr>
              <td class="smallbold">URL</td>
              <td>http://httpbin.org/post</td>
            </tr>
            <tr>
              <td class="smallbold">Request Method</td>
              <td>
                <span class="d-flex text-left align-items-center">POST</span>
              </td>
            </tr>
            <tr>
              <td class="smallbold">Status Code</td>
              <td><span class="badge badge-success">200</span></td>
            </tr>
          </tbody>
        </table>
        <hr class="my-3" />
        <div class="matching-details mx-3">
          <h3 class="my-2">Match Information</h3>
          <p class="text-text-light">Nothing matched</p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #1`] = `
<div
  data-role="log-details-modal"
  class="modal-dialog modal-lg flow-detail"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Secure Data</h5>
      <span id="" class="cursor-pointer"><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <nav class="nav-tabs nav-tabs-sm">
        <a href="#" class="capitalize" data-role="tab-general">general</a
        ><a href="#" class="active capitalize" data-role="tab-headers"
          >headers</a
        ><a href="#" class="capitalize" data-role="tab-body">body</a>
      </nav>
      <div class="text-center mt-3 mb-2 position-relative">
        <button
          type="button"
          class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
        >
          <span>Secure this payload</span>
        </button>
        <div role="group" class="btn-group">
          <button
            type="button"
            data-role="select-request-phase"
            class="active capitalize btn btn-primary-light btn-sm"
          >
            request</button
          ><button
            type="button"
            data-role="select-response-phase"
            class="capitalize btn btn-outline btn-sm"
          >
            response
          </button>
        </div>
        <hr class="my-3 p-0" />
      </div>
      <div class="mt-3 pt-0">
        <section class="request mt-2">
          <table class="header-table table table-striped">
            <tbody>
              <tr class="">
                <td class="header-name">Host</td>
                <td class="header-original">httpbin.org</td>
              </tr>
              <tr class="">
                <td class="header-name">User-Agent</td>
                <td class="header-original">curl/7.64.1</td>
              </tr>
              <tr class="">
                <td class="header-name">Accept</td>
                <td class="header-original">*/*</td>
              </tr>
              <tr class="">
                <td class="header-name">Proxy-Connection</td>
                <td class="header-original">Keep-Alive</td>
              </tr>
              <tr class="">
                <td class="header-name">Content-Type</td>
                <td class="header-original">application/json</td>
              </tr>
              <tr class="">
                <td class="header-name">Content-Length</td>
                <td class="header-original">14</td>
              </tr>
            </tbody>
          </table>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #2`] = `
<div
  data-role="log-details-modal"
  class="modal-dialog modal-lg flow-detail"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Secure Data</h5>
      <span id="" class="cursor-pointer"><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <nav class="nav-tabs nav-tabs-sm">
        <a href="#" class="capitalize" data-role="tab-general">general</a
        ><a href="#" class="active capitalize" data-role="tab-headers"
          >headers</a
        ><a href="#" class="capitalize" data-role="tab-body">body</a>
      </nav>
      <div class="text-center mt-3 mb-2 position-relative">
        <button
          type="button"
          class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
        >
          <span>Secure this payload</span>
        </button>
        <div role="group" class="btn-group">
          <button
            type="button"
            data-role="select-request-phase"
            class="capitalize btn btn-outline btn-sm"
          >
            request</button
          ><button
            type="button"
            data-role="select-response-phase"
            class="active capitalize btn btn-primary-light btn-sm"
          >
            response
          </button>
        </div>
        <hr class="my-3 p-0" />
      </div>
      <div class="mt-3 pt-0">
        <section class="request mt-2">
          <table class="header-table table table-striped">
            <tbody>
              <tr class="">
                <td class="header-name">Date</td>
                <td class="header-original">Wed, 26 Feb 2020 15:22:46 GMT</td>
              </tr>
              <tr class="">
                <td class="header-name">Content-Type</td>
                <td class="header-original">application/json</td>
              </tr>
              <tr class="">
                <td class="header-name">Content-Length</td>
                <td class="header-original">464</td>
              </tr>
              <tr class="">
                <td class="header-name">Connection</td>
                <td class="header-original">keep-alive</td>
              </tr>
              <tr class="">
                <td class="header-name">Server</td>
                <td class="header-original">gunicorn/19.9.0</td>
              </tr>
              <tr class="">
                <td class="header-name">Access-Control-Allow-Origin</td>
                <td class="header-original">*</td>
              </tr>
              <tr class="">
                <td class="header-name">Access-Control-Allow-Credentials</td>
                <td class="header-original">true</td>
              </tr>
            </tbody>
          </table>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #3`] = `
<div
  data-role="log-details-modal"
  class="modal-dialog modal-lg flow-detail"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Secure Data</h5>
      <span id="" class="cursor-pointer"><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <nav class="nav-tabs nav-tabs-sm">
        <a href="#" class="capitalize" data-role="tab-general">general</a
        ><a href="#" class="capitalize" data-role="tab-headers">headers</a
        ><a href="#" class="active capitalize" data-role="tab-body">body</a>
      </nav>
      <div class="text-center mt-3 mb-2 position-relative">
        <button
          type="button"
          class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
        >
          <span>Secure this payload</span>
        </button>
        <div role="group" class="btn-group">
          <button
            type="button"
            data-role="select-request-phase"
            class="active capitalize btn btn-primary-light btn-sm"
          >
            request</button
          ><button
            type="button"
            data-role="select-response-phase"
            class="capitalize btn btn-outline btn-sm"
          >
            response
          </button>
        </div>
        <hr class="my-3 p-0" />
      </div>
      <div class="mt-3 pt-0">
        <div class="mb-3" data-role="log-payload-body-diff">
          <div class="diffsnippet">
            <table class="css-13qjma8-diff-container">
              <tbody>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>1</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>{"foo": "bar"}</pre>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #4`] = `
<div
  data-role="log-details-modal"
  class="modal-dialog modal-lg flow-detail"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Secure Data</h5>
      <span id="" class="cursor-pointer"><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <nav class="nav-tabs nav-tabs-sm">
        <a href="#" class="capitalize" data-role="tab-general">general</a
        ><a href="#" class="capitalize" data-role="tab-headers">headers</a
        ><a href="#" class="active capitalize" data-role="tab-body">body</a>
      </nav>
      <div class="text-center mt-3 mb-2 position-relative">
        <button
          type="button"
          class="ant-btn d-flex position-absolute right-0 ant-btn-primary ant-btn-sm"
        >
          <span>Secure this payload</span>
        </button>
        <div role="group" class="btn-group">
          <button
            type="button"
            data-role="select-request-phase"
            class="capitalize btn btn-outline btn-sm"
          >
            request</button
          ><button
            type="button"
            data-role="select-response-phase"
            class="active capitalize btn btn-primary-light btn-sm"
          >
            response
          </button>
        </div>
        <hr class="my-3 p-0" />
      </div>
      <div class="mt-3 pt-0">
        <div class="mb-3" data-role="log-payload-body-diff">
          <div class="diffsnippet">
            <table class="css-13qjma8-diff-container">
              <tbody>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>1</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed"><pre>{</pre></td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>2</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "args": {}, </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>3</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "data": "{\\"foo\\": \\"bar\\"}", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>4</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "files": {}, </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>5</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "form": {}, </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>6</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "headers": {</pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>7</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>    "Accept": "*/*", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>8</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>    "Content-Length": "14", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>9</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>    "Content-Type": "application/json", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>10</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>    "Host": "httpbin.org", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>11</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>    "Proxy-Connection": "Keep-Alive", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>12</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>    "User-Agent": "curl/7.64.1", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>13</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>
    "X-Amzn-Trace-Id": "Root=1-5e568d46-121257eeb4ab060a1c8591c0"</pre
                    >
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>14</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed"><pre>  }, </pre></td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>15</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "json": {</pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>16</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>    "foo": "bar"</pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>17</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed"><pre>  }, </pre></td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>18</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "origin": "46.219.215.255", </pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>19</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed">
                    <pre>  "url": "http://httpbin.org/post"</pre>
                  </td>
                </tr>
                <tr class="css-1j7ahnq-line">
                  <td class="css-5o05yy-gutter css-1h3hzrb-diff-removed">
                    <pre>20</pre>
                  </td>
                  <td
                    class="css-5o05yy-gutter css-1klnsbn-empty-gutter css-1h3hzrb-diff-removed"
                  >
                    <pre></pre>
                  </td>
                  <td class="css-1u7u2ow-marker css-1h3hzrb-diff-removed">
                    <pre>-</pre>
                  </td>
                  <td class="css-1h3hzrb-diff-removed"><pre>}</pre></td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Localhoste upload single har flow > Visits Localhoste and gets 2 yamls from uploaded har file #5`] = `
<div
  data-role="quick-integration-modal"
  class="modal-dialog modal-lg"
  role="document"
>
  <div class="modal-content">
    <div class="modal-header">
      <h5 class="modal-title">Secure a Payload</h5>
      <span id="" class="cursor-pointer"><i class="fa-times fas"></i></span>
    </div>
    <div class="modal-body">
      <div>
        <h4 class="mb-3">
          Select the fields that you want to Redact<sup
            ><i
              aria-label="icon: question-circle"
              tabindex="-1"
              class="anticon anticon-question-circle d-inline-flex mr-2 text-text text-sm"
              style="font-size: 14px;"
              ><svg
                viewBox="64 64 896 896"
                focusable="false"
                class=""
                data-icon="question-circle"
                width="1em"
                height="1em"
                fill="currentColor"
                aria-hidden="true"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
                ></path></svg></i></sup
          >or Reveal<sup
            ><i
              aria-label="icon: question-circle"
              tabindex="-1"
              class="anticon anticon-question-circle d-inline-flex mr-2 text-text text-sm"
              style="font-size: 14px;"
              ><svg
                viewBox="64 64 896 896"
                focusable="false"
                class=""
                data-icon="question-circle"
                width="1em"
                height="1em"
                fill="currentColor"
                aria-hidden="true"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 0 0-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
                ></path></svg></i
          ></sup>
        </h4>
        <div>
          <div>
            <ul class="css-5q6xdk">
              <li class="css-79elbk">
                <div>
                  <input name="isGoing" type="checkbox" readonly="" />
                  <div
                    title="$.foo"
                    data-role="checkbox-text"
                    class="w-95 text-word-break"
                    style="display: inline-block; vertical-align: top; color: inherit; padding-left: 10px; cursor: pointer;"
                  >
                    foo: bar
                  </div>
                </div>
                <div><ul class="css-1s1o8u8" style="display: block;"></ul></div>
              </li>
            </ul>
            <div class="mt-5 text-center">
              <button
                type="button"
                disabled=""
                data-role="select-secure-payload"
                class="button-next btn btn-primary disabled"
              >
                Secure this payload
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
